### PR TITLE
OTel-Arrow receiver: support AuthServer without IncludeMetadata

### PR DESCRIPTION
**Pull request by jmacd**
**Description:** 
Ensure the Auth Server implementation receives the complete set of headers, even when IncludeMetadata is not set. Prior to this fix, the AuthServer called per stream-data-item would only function when IncludeMetadata was set. IncludeMetadata determines what the pipeline sees, but the AuthServer should see the headers either way.

**Link to tracking Issue:** #45.

### CODE REVIEW

File: receiver/otlpreceiver/internal/arrow/arrow.go [hunk 1]
```diff
@@ -135,64 +146,85 @@ func newHeaderReceiver(streamCtx context.Context, includeMetadata bool) *headerR
 // client.Info with additional key:values associated with the arrow batch.
 // This is safe to call when h is nil.
```
Comment (jaronoff97) [inline receiver/otlpreceiver/internal/arrow/arrow.go:147]: It looks like we're removing the nil check on `h` here, is that okay to do?

File: receiver/otlpreceiver/internal/arrow/arrow_test.go [hunk 2]
```diff
@@ -719,6 +720,38 @@ func TestHeaderReceiverNoIncludeMetadata(t *testing.T) {
 	}
 }
 
+func TestHeaderReceiverAuthServerNoIncludeMetadata(t *testing.T) {
```
Comment (jaronoff97) [inline receiver/otlpreceiver/internal/arrow/arrow_test.go:723]: quibble: could we rename this test to be `TestHeaderReceiverAuthServerIsSetNoIncludeMetadata` or something similar? initially my brain read this as testing the header receiver's auth server when include metadata is false and not testing that when the auth server isn't null and include is false that we indeed receive the headers expected.

File: receiver/otlpreceiver/internal/arrow/arrow.go [hunk 3]
```diff
@@ -135,64 +146,85 @@ func newHeaderReceiver(streamCtx context.Context, includeMetadata bool) *headerR
...
-		// extension's GetRequestMetadata().  This paired with the
-		// headersetter's return of empty-string values means, we would
-		// end up with an empty-string element for any headersetter
-		// `from_context` rules b/c the stream uses background context.
-		// This allows static headers through.
-		//
-		// See https://github.com/open-telemetry/opentelemetry-collector/issues/6965
-		if _, ok := h.tmpHdrs[k]; !ok {
-			h.tmpHdrs[k] = v
+	if h.tmpHdrs != nil {
```
Comment (jaronoff97) [inline receiver/otlpreceiver/internal/arrow/arrow.go:171]: given that checking the existence of a key in a nil map will [always return false](https://go.dev/play/p/sk-vjkwRx6e), are we doing this nil check to short circuit so we don't iterate over the stream headers unnecessarily?

File: receiver/otlpreceiver/internal/arrow/arrow.go (hunk 1 above)
Comment (jmacd) [reply receiver/otlpreceiver/internal/arrow/arrow.go:147]: Right, there is no longer a nil in this setting. Fixed the comment.

**Review by jaronoff97 — APPROVED** [review]
looks great 🎉

**Review by lquerel — APPROVED** [review]
LGTM
